Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -631,6 +654,41 @@ void GameList::OpenGCSaveFolder()
     ModalMessageBox::information(this, tr("Information"), tr("No save data found."));
 }
 
+#ifdef _WIN32
+bool GameList::AddShortcutToDesktop()
+{
+  auto init = wil::CoInitializeEx_failfast(COINIT_APARTMENTTHREADED);
```

Comment:
Not sure if `failfast` is desirable here - COM may already be initialized in a different threading model, and I don't think this code **must** use an apartment threaded model?

Reply:
That is (I think?) the only wrapper provided by WIL. Multithreaded would not work here as it's on a GUI thread. Specifying multithreaded resulted in an assert, so I think it's working as intended? However, I didn't investigate further to find out what is calling CoInitialize on this thread beforehand.

All CoInitialize* calls on a specific thread _should_ specify the same threading model, even though it's possible to specify multiple threading models (and ignore the returned error code). I think it's better to be explicit about the expectation (which is probably why WIL only exposes a _failfast version of the API).

Reply:
It's possible - but if you don't want to make a strong assumption like this, you could. My WASAPI PR realizes this by calling `CoInitializeEx` "manually" and then activating/deactivating the RAII `CoUnitialize` call basing on the result.

Reply:
I noticed [cubeb has something similar](https://github.com/dolphin-emu/dolphin/blob/0029ca84b07f44e3ce8c80a44fa0285a966b34d3/Externals/cubeb/src/cubeb_wasapi.cpp#L138). I'd rather use what's provided by WIL, and I don't think making a strong assumption is a problem here. Ideally, all usages of CoInitialize would make strong assumptions (therefor if they trigger fastfail, it's a good indication some code is actually being used incorrectly).
If you want to make a wrapper like cubeb's, maybe consider moving it to some more common location in dolphin?